cmds/core/shutdown/shutdown_linux.go

```diff
@@ -41,48 +41,54 @@ var (
 		"suspend": unix.LINUX_REBOOT_CMD_SW_SUSPEND,
 		"-s":      unix.LINUX_REBOOT_CMD_SW_SUSPEND,
 	}
-	reboot = unix.Reboot
-	delay  = time.Sleep
 )
 
 func usage() {
-	log.Fatalf("shutdown [<-h|-r|-s|halt|reboot|suspend> [time [message...]]]")
+	fmt.Println("shutdown [<-h|-r|-s|halt|reboot|suspend> [time [message...]]]")
 }
 
-func main() {
-	a := os.Args
-	if len(a) == 1 {
-		a = append(a, "halt")
+func shutdown(args []string) (uint, error) {
+	if len(args) == 1 {
+		args = append(args, "halt")
 	}
-	op, ok := opcodes[a[1]]
+	op, ok := opcodes[args[1]]
 	if !ok {
 		usage()
+		return 0, nil
 	}
-	if len(a) < 3 {
-		a = append(a, "now")
+	if len(args) < 3 {
+		args = append(args, "now")
 	}
 	when := time.Now()
 	switch {
-	case a[2] == "now":
-	case a[2][0] == '+':
-		m, err := time.ParseDuration(a[2][1:] + "m")
+	case args[2] == "now":
+
+	case args[2][0] == '+':
+		m, err := time.ParseDuration(args[2][1:] + "m")
 		if err != nil {
-			log.Fatal(err)
+			return 0, err
 		}
 		when = when.Add(m)
 	default:
-		t, err := time.Parse(time.RFC3339, a[2])
+		t, err := time.Parse(time.RFC3339, args[2])
 		if err != nil {
-			log.Fatal(err)
+			return 0, err
 		}
 		when = t
 	}
 
-	delay(when.Sub(time.Now()))
-	if len(a) > 2 {
-		fmt.Println(strings.Join(a[3:], " "))
+	time.Sleep(time.Until(when))
+	if len(args) > 2 {
+		fmt.Println(strings.Join(args[3:], " "))
+	}
+	if err := unix.Reboot(int(op)); err != nil {
+		return 0, err
 	}
-	if err := reboot(int(op)); err != nil {
+	return op, nil
+}
+
+func main() {
+	if _, err := shutdown(os.Args); err != nil {
```

Comment:
you might change this:
shutdown(os.Args[1:]...)
see above.

Reply:
Cant mix os.Args[1:] and ...string